src/deluge/storage/flash_storage.cpp

```diff
@@ -420,6 +432,44 @@ void readSettings() {
 		defaultMetronomeVolume = kMaxMenuMetronomeVolumeValue;
 	}
 	AudioEngine::metronome.setVolume(defaultMetronomeVolume);
+
+	/* Future bytes reserved for PR #781, MIDI follow */
+	/* 123 -> 131 */
+	/*
+	midiEngine.midiFollow = buffer[123];
+	midiEngine.midiFollowChannelSynth = buffer[124];
+	midiEngine.midiFollowChannelKit = buffer[125];
+	midiEngine.midiFollowChannelParam = buffer[126];
+	midiEngine.midiFollowKitRootNote = buffer[127];
+	midiEngine.midiFollowDisplayParam = buffer[128];
+	midiEngine.midiFollowFeedback = buffer[129];
+	midiEngine.midiFollowFeedbackAutomation = static_cast<MIDIFollowFeedbackAutomationMode>(buffer[130]);
+	midiEngine.midiFollowFeedback = buffer[131];
+	*/
+
+	/* Bytes 132-134 have migrated from a previous location. Loading firmware either side
+	   of the change sequentially can risk reading arbitrary data, which will
+	   cast to an enum of the right type, but with a value that matches none of
+	   the available enum values. Pick a valid default if this happens. */
+	defaultSessionLayout = static_cast<SessionLayoutType>(buffer[132]);
```

Comment:
I think we should test these with a pattern like

```cpp
if (util::to_underlying(SessionLayoutTypeMaxElement) <= buffer[132]) {
  defaultSessionLayout = SessionLayoutType::SessionLayoutTypeRows;
} else {
  defaultSessionLayout = static_cast<SessionLayoutType>(buffer[132]);
}
```

This way we avoid ever having bit patterns that represent invalid enum values in a variable with that enum's type.

Reply:
Shall I apply this to the other enum class based flash settings that go further back? These have all worked OK in the past, just defaultSessionLayout and defaultKeyboardLayout are vulnerable, especially if you switch between firmwares that straddle the branch to 1.0.1

Reply:
Up to you. My preference is slightly towards fixing it for all cases, but we can likely trust it won't misbehave in practice for anyone but developers for those but it's a potential problem vector.